File: src/main/java/app/application/listener/YoutubeVideoDownloadListener.java
package app.application.listener;

import app.application.utils.DialogManager;
import javafx.application.Platform;
import javafx.scene.control.ProgressBar;

import java.io.File;

public class YoutubeVideoDownloadListener extends YoutubeDownloadListener {

    public static final int FINISHED_PROGRESS = 100;
    private final ProgressBar progressBar;

    private int currentProgress;

    public YoutubeVideoDownloadListener(ProgressBar progressBar, DialogManager dialogManager) {
        super(dialogManager);
        this.progressBar = progressBar;
    }

    @Override
    public void onFinished(File data) {
        Platform.runLater(() -> dialogManager.openInformationDialog("Download fertig.", ""));
    }

    @Override
    public void onDownloading(int progress) {
        this.currentProgress = progress;
        Platform.runLater(() -> progressBar.setProgress(Double.parseDouble(Integer.toString(progress)) / 100));
    }

    @Override
    public boolean isDownloadFinished() {
        return currentProgress >= FINISHED_PROGRESS;
    }

}

File: src/main/java/app/application/listener/YoutubeVideoDownloadListener.kt
package app.application.listener

import app.application.utils.DialogManager
import app.application.utils.GlobalObjectHandler
import javafx.application.Platform
import javafx.scene.control.ProgressBar
import java.io.File

class YoutubeVideoDownloadListener(private val progressBar: ProgressBar, dialogManager: DialogManager, private val globalObjectHandler: GlobalObjectHandler) : YoutubeDownloadListener(dialogManager) {
    private var currentProgress = 0
    override fun onFinished(data: File) {
        globalObjectHandler.hostServices!!.showDocument(data.parent)
        Platform.runLater { dialogManager!!.openInformationDialog("Download fertig.", "") }
    }

    override fun onDownloading(progress: Int) {
        currentProgress = progress
        Platform.runLater { progressBar.progress = Integer.toString(progress).toDouble() / 100 }
    }

    override val isDownloadFinished: Boolean
        get() = currentProgress >= FINISHED_PROGRESS

    companion object {
        const val FINISHED_PROGRESS = 100
    }
}